src/components/utils/src/threads/async_runner.cc

```diff
@@ -54,12 +54,12 @@ void AsyncRunner::AsyncRun(ThreadDelegate* delegate) {
 
 void AsyncRunner::Stop() {
   SDL_AUTO_TRACE();
-  thread_->join();
+  thread_->join(true);
 }
 
 AsyncRunner::~AsyncRunner() {
   SDL_AUTO_TRACE();
-  thread_->join();
+  thread_->join(true);
   delete executor_;
```

Comment:
@orhan-mehmedov just a question to be in the clear, why dont you call the delegate() method here on executor_ ?